wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift

```diff
@@ -103,6 +103,24 @@ public final class ConversationService: ConversationServiceInterface {
             return
         }
 
+        func retry(excludingDomains domains: Set<String>) {
```

Comment:
There's a lot of duplication here, one way we could avoid it is something like:

```swift
func createGroup(
    withUsers users: Set<ZMUser>, 
    completion: @escaping (Swift.Result<ZMConversation, ConversationCreationFailure>) -> Void
) {
    internalCreateGroupConversation(
        teamID: teamID,
        name: name,
        users: usersToAdd,
        failedToAddUsers: usersToExclude,
        accessMode: ConversationAccessMode.value(forAllowGuests: allowGuests),
        accessRoles: ConversationAccessRoleV2.from(
            allowGuests: allowGuests,
            allowServices: allowServices
        ),
        enableReceipts: enableReceipts,
        messageProtocol: messageProtocol,
        completion: completion
    )
}

createGroup(withUsers: users) { result in
    switch result {
    case .failure(.networkError(.unreachableDomains(let domains))):
        let usersToExclude = users.belongingTo(domains: domains)
        let reachableUsers = Set(users).subtracting(usersToExclude)
        createGroup(withUsers: reachableUsers, completion: completion)

    default:
        completion(result)
    }
}
```

Reply:
Updated